hail/src/main/scala/is/hail/expr/ir/functions/NDArrayFunctions.scala

```diff
@@ -35,6 +36,53 @@ object NDArrayFunctions extends RegistryFunctions {
         NDArrayMap2(l, r, lid, rid, irOp(lElemRef, rElemRef, errorID), errorID)
       }
     }
+    def linear_triangular_solve(ndCoef: SNDArrayCode, ndDep: SNDArrayCode, lower: SBooleanCode, outputPt: PType, cb: EmitCodeBuilder, region: Value[Region], errorID: Value[Int]): (SNDArrayCode, Value[Int]) = {
+      val ndCoefInput = ndCoef.asNDArray.memoize(cb, "ndCoef")
+      val ndDepInput = ndDep.asNDArray.memoize(cb, "ndDep")
+
+      val ndCoefColMajor = LinalgCodeUtils.checkColMajorAndCopyIfNeeded(ndCoefInput, cb, region)
+      val ndDepColMajor = LinalgCodeUtils.checkColMajorAndCopyIfNeeded(ndDepInput, cb, region)
+
+      val IndexedSeq(ndCoefRow, ndCoefCol) = ndCoefColMajor.shapes(cb)
+      cb.ifx(ndCoefRow cne ndCoefCol, cb._fatalWithError(errorID, "hail.nd.solve_triangular: matrix a must be square."))
+
+      val IndexedSeq(ndDepRow, ndDepCol) = ndDepColMajor.shapes(cb)
+      cb.ifx(ndCoefRow  cne ndDepRow, cb._fatalWithError(errorID,"hail.nd.solve_triangular: Solve dimensions incompatible"))
+
+      val uplo = cb.newLocal[String]("dtrtrs_uplo")
+      cb.ifx(lower.boolCode(cb), cb.assign(uplo, const("L")), cb.assign(uplo, const("U")))
+
+      val infoDTRTRSResult = cb.newLocal[Int]("dtrtrs_result")
+
+      val trans = cb.newLocal[String]("dtrtrs_trans")
+      cb.assign(trans, const("N"))
+
+      val diag = cb.newLocal[String]("dtrtrs_diag")
+      cb.assign(diag, const("N"))
```

Comment:
No need to make values for these. You are only going to use `trans` and `diag` once.